core/src/main/scala/kafka/common/ZkNodeChangeNotificationListener.scala

```diff
@@ -58,6 +59,7 @@ class ZkNodeChangeNotificationListener(private val zkUtils: ZkUtils,
   def init() {
     zkUtils.makeSurePersistentPathExists(seqNodeRoot)
     zkUtils.zkClient.subscribeChildChanges(seqNodeRoot, NodeChangeListener)
+    zkUtils.zkClient.subscribeStateChanges(ZkStateChangeListener)
     processAllNotifications()
```

Comment:
> ZkStateChangeListener'handleNewSession method will only be called if a reconnection happens or if a child changes

A child change won't cause handleNewSession to be called, and a reconnection is not supposed to induce a call to it either. It needs to be a new session as I understand it. A reconnection would be signaled by the state change method.

> To avoid that delay we call the processAllNotifications() as part of init.

It sounds like this is going to induce duplicate requests to ZK, is the processing of this notifications in the case of duplicate calls idempotent?